molter/command.py

```diff
@@ -741,3 +789,41 @@ def wrapper(func: Callable) -> MolterCommand:
 
 
 msg_command = message_command
+
+# molter command typevar - can be the function or the command
+MCT = TypeVar("MCT", Callable, MolterCommand)
+
+
+def register_converter(anno_type: type, converter: type[Converter]) -> Callable[..., MCT]:
+    """
+    A decorator that allows you to register converters for a type for a specific command.
+
+    This allows for native type annotations without needing to use `typing.Annotated`.
+
+    Args:
+        anno_type (`type`): The type to register for.
+        converter (`type[Converter]`): The converter to use for the type.
+
+    Returns:
+        `Callable | MolterCommand`: Either the callback or the command.
+        If this is used after using the `molter.message_command` decorator, it will be a command.
+        Otherwise, it will be a callback.
+    """
+
+    def wrapper(command: MCT) -> MCT:
+        if hasattr(command, "_type_to_converter"):
+            command._type_to_converter[anno_type] = converter
+        else:
+            command._type_to_converter = {anno_type: converter}
+
+        if isinstance(command, MolterCommand):
+            # yes, this sucks, but we need to re-analyze params just to make sure all
+            # annotations get converted with the new converter injection
+            # when this is merged with dis-snek, parameters will be analyzed either
+            # when the client reads its own commands or when a scale is loaded,
+            # instead of on creation. that will remove the need for this
+            command.parse_parameters(has_self=_is_nested(command.callback))
```

Comment:
I'll review myself SMH.

This section is the only section I have concerns with. I would like to do a less naive approach where we go through each parameter and only redo the ones that are needed here.